DeleteUserDataActivity/handler.ts

```diff
@@ -0,0 +1,617 @@
+ * @param lazyPromise a lazy promise to convert
+ * @param queryName an optional name for the query, for logging purpose
+ *
+ * @returns either the query result or a query failure
+ */
+const fromQueryEither = <R>(
+  lazyPromise: () => Promise<Either<QueryError | Error, R>>
+): TaskEither<Error, R> =>
+  tryCatch(lazyPromise, toError).chain(errorOrResult =>
+    fromEither(errorOrResult).mapLeft((err: QueryError | Error) =>
+      err instanceof Error
+        ? err
+        : new Error(`QueryError: ${JSON.stringify(err)}`)
+    )
+  );
+
+/**
+ * To be used for exhaustive checks
+ */
+function assertNever(_: never): void {
+  throw new Error("should not have executed this");
+}
+
+/**
+ * to cast an error to QueryFailure
+ * @param err
+ */
+const toQueryFailure = (err: Error | QueryError): QueryFailure =>
+  QueryFailure.encode({
+    kind: "QUERY_FAILURE",
+    reason:
+      err instanceof Error ? err.message : `QueryError: ${JSON.stringify(err)}`
+  });
+
+/**
+ * to cast an error to a DocumentDeleteFailure
+ * @param err
+ */
+const toDocumentDeleteFailure = (err: Error): DocumentDeleteFailure =>
+  DocumentDeleteFailure.encode({
+    kind: "DELETE_FAILURE",
+    reason: err.message
+  });
+
+/**
+ * Logs depending on failure type
+ * @param context the Azure functions context
+ * @param failure the failure to log
+ */
+const logFailure = (context: Context) => (
+  failure: ActivityResultFailure
+): void => {
+  switch (failure.kind) {
+    case "INVALID_INPUT_FAILURE":
+      context.log.error(
+        `${logPrefix}|Error decoding input|ERROR=${failure.reason}`
+      );
+      break;
+    case "QUERY_FAILURE":
+      context.log.error(
+        `${logPrefix}|Error ${failure.query} query error|ERROR=${failure.reason}`
+      );
+      break;
+    case "BLOB_FAILURE":
+      context.log.error(
+        `${logPrefix}|Error saving zip bundle|ERROR=${failure.reason}`
+      );
+      break;
+    case "USER_NOT_FOUND_FAILURE":
+      context.log.error(`${logPrefix}|Error user not found|ERROR=`);
+      break;
+    case "DELETE_FAILURE":
+      context.log.error(
+        `${logPrefix}|Error deleting data|ERROR=${failure.reason}`
+      );
+      break;
+    default:
+      assertNever(failure);
+  }
+};
+
+// define a value object with the info related to the blob storage for backup files
+interface IBlobServiceInfo {
+  blobService: BlobService;
+  containerName: string;
+  folder?: NonEmptyString;
+}
+
+/**
+ * Saves data into a dedicated blob
+ * @param blobServiceInfo references about where to save data
+ * @param blobName name of the blob to be saved. It might not include a folder if specified in blobServiceInfo
+ * @param data serializable data to be saved
+ *
+ * @returns either a blob failure or the saved object
+ */
+export const saveDataToBlob = <T>(
+  { blobService, containerName, folder }: IBlobServiceInfo,
+  blobName: string,
+  data: T
+): TaskEither<BlobCreationFailure, T> =>
+  taskify(blobService.createBlockBlobFromText)(
```

Comment:
```suggestion
  taskify(blobService.createBlockBlobFromText.bind(blobService)(
```

we don't know if `createBlockBlobFromText` is using a referente to '_this_' (probably not, but since we cannot know we have to bind the method anyway)